dev-games/godot/godot-3.2.3.ebuild

```diff
@@ -0,0 +1,136 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+PYTHON_COMPAT=( python3_{6..9} )
+
+inherit eutils python-any-r1 scons-utils flag-o-matic desktop xdg
+
+DESCRIPTION="Multi-platform 2D and 3D game engine"
+HOMEPAGE="http://godotengine.org"
+
+SRC_URI="https://github.com/godotengine/${PN}/archive/${PV}-stable.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/${P}-stable"
+KEYWORDS="~amd64 ~x86"
+
+LICENSE="MIT"
```

Comment:
You should also include licenses for the [Godot logo](https://github.com/godotengine/godot/blob/master/LOGO_LICENSE.md) and the [third-party code](https://github.com/godotengine/godot/blob/master/thirdparty/README.md) used (assuming those are part of the installed files).

Reply:
The [Debian godot package](https://packages.debian.org/sid/godot3) offers a good starting point to locating all the licenses in Godot: https://metadata.ftp-master.debian.org/changelogs//main/g/godot/godot_3.2-stable-2_copyright

Reply:
this ebuild has disabled building most of the 3rd party libs and link with the ones provided by gentoo's tree instead, so it might not be necessary. But I will add the logo's license nontheless.

Reply:
If the code under the [`thirdparty`](https://github.com/godotengine/godot/tree/31d0f8ad8d5cf50a310ee7e8ada4dcdb4510690b/thirdparty) is not used, you should remove this directory in `src_prepare()` to ensure that only system libraries are used:

```
src_prepare() {
        # Remove bundled libaries
	rm -rf thirdparty || die
}
```

In addition, those third party libraries don't appear to be the only code with different licenses. For example, [`servers/physics/joints/cone_twist_joint_sw.cpp`](https://github.com/godotengine/godot/blob/31d0f8ad8d5cf50a310ee7e8ada4dcdb4510690b/servers/physics/joints/cone_twist_joint_sw.cpp) is licensed under **zlib** license too.

To be fair, I only see **zlib** as another license in the main Godot code in my quick glace, so adding just that and the icon license might suffice for this ebuild if the thirdparty directory is removed during `src_prepare()`.

Reply:
I might have overlooked this one. It turns out assimp is still required as a built-in library.

Reply:
That's all right, just remove the directories within `thirdparty` that aren't needed, and include the licenses for the remaining that are used.

Reply:
There are too many of the directories in there and it certainly takes time to list all of the unneeded directories one by one.

Reply:
Unfortunately, proper documentation of code licenses is important, so that's part of the plight of being an ebuild maintainer. Luckily, there might be a way to automate this a bit with some bash magic.

First remove the directories for the libraries you know aren't used because you're pulling in the system libraries. Then use [`find`](https://man7.org/linux/man-pages/man1/find.1.html) to recursively get all the `COPYRIGHT` and `LICENSE` files, and pass them to [`head`](https://man7.org/linux/man-pages/man1/head.1.html) to print out the first few lines. Finally you can just scroll down the lines printed to see all the license names of the code remaining. :-)

Reply:
Alright, I have deleted unused third party libs and enumerated all the licenses for the remaining libs. It should be no longer a problem by now.